designs/package-spec.md

```diff
@@ -0,0 +1,78 @@
+# Package Controller Configuration
+
+## Introduction
+
+**Problem:** currently the ability to configure the package controller is very limited and requires code changes
+
+Code changes currently have to be made to configure the package controller during cluster creation or using the install command. Changes need to be coordinated between the CLI and the package controller helm chart to alter configuration. This makes it more difficult to support things like various private registry configurations.
+
+### Tenets
+
+* ***Flexible:*** reduce dependency of the CLI on helm chart.
+
+### Goals and Objectives
+
+As an EKS-A cluster administrator I want to configure the package controller for:
+
+* Harbor using proxy cache
+* Harbor using replication
+* Harbor as a private registry using the import images command
+* Configure ECR token refresher
+* Configure for kubelet private registry authentication
+* Support proxy configuration
+* Support package controller installation from alternate registries
+* Disable installation of package controller
+* Allow for future configuration options
+
+### Statement of Scope
+
+**In scope**
+* All of the above
+
+## Current state
+The package controller can be used currently with:
+1. Proxy configurations
+1. ECR token refresher enabled/disabled
+1. Installation of the package controller from alternate registries
+
+## Overview of Solution
+Add a section to the cluster specification to enable/disable the package controller installation and pass configuration values.
+
+## Solution Details
+
+Add a packageController section to the cluster specification for example to use development/staging builds:
```

Comment:
I would hope all these values would be optional and default to not set. This is for power users.

As for registry mirror configuration, I think it makes sense to possibly pull packages from a registry mirror while EKS-A is sourced from another mirror or ECR. I think if a registry mirror is set here, it would override the mirror from EKS-A spec.

Reply:
As to preflight, including these values in the cluster spec will do some validation preflight.